### ISSUE TITLE
Crashes if given an OTIOZ

### ISSUE THREAD

**Issue by jminor** [OpenTimelineIO/raven#79]
If you launch raven with an OTIOZ, it crashes.

```
(base) ~⭐ raven ALab_OTIO/ALab_mk020_final_edit.mxf.relinked.otioz
LOG: Error loading "ALab_OTIO/ALab_mk020_final_edit.mxf.relinked.otioz": JSON parse error: JSON parse error on input string: Invalid value. (line 1, column 0)
2024-10-24 10:11:05.292 raven[2185:9649528] +[IMKClient subclass]: chose IMKClient_Legacy
2024-10-24 10:11:05.292 raven[2185:9649528] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
zsh: segmentation fault  raven ALab_OTIO/ALab_mk020_final_edit.mxf.relinked.otioz
```

**Comment by jminor** [comment]
At bare minimum, it should report a better error message and exit without crashing. Of course, it would be even better, if it could read the OTIO inside the OTIOZ.

**Comment by ThomasWilshaw** [comment]
I know the OTIO Python API can open OTIOZ files but am I right in thinking the cpp API can't? Is that something that needs to be added there or would you rather it was implemented in Raven?

**Comment by jminor** [comment]
There was some recent discussion on this topic here: https://academysoftwarefdn.slack.com/archives/CMQ9J4BQC/p1729878665473649

It would be great if the core OTIO C++ SDK had support for OTIOZ, but it currently doesn't. Maybe we could implement this in Raven just to see what minimal API is needed, and then port that into the OTIO core? From the Slack thread linked above, it seems like there are several candidates for which ZIP archive library to use.

**Comment by ThomasWilshaw** [comment]
Thanks for the link, as you say there seems to be various options. Given OCIO uses [minizip-ng](https://github.com/zlib-ng/minizip-ng) that would seem a sensible place to start? I suppose Raven would only want to extract the OTIO file and avoid having to extract the entire archive, whereas OTIO itself would maybe want two finctions, `extract_otio_from_otioz` and `unzip_otioz`?

Some possibly useful links for reference:

[OCIO zip implementation](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/src/OpenColorIO/OCIOZArchive.cpp#L335)
[minizip-ng extract single file example](https://github.com/zlib-ng/minizip-ng/blob/develop/minizip.c#L351)

**Comment by jminor** [comment]
You're right that Raven only needs to read the `content.otio` from the OTIOZ, not the media. Ideally it would read it directly into memory without needing to write anything to disk.

Here's a first attempt: https://github.com/OpenTimelineIO/raven/pull/85

@ThomasWilshaw I'd love a code review and/or testing of that PR if you've got the time.
